plugins/statistics.koplugin/main.lua

```diff
@@ -2304,13 +2332,40 @@ function ReaderStatistics:onSuspend()
     if not self:isDocless() then
         self.ui.doc_settings:saveSetting("stats", self.data)
         self:insertDB()
+        self:onReadingPaused()
     end
 end
 
 -- screensaver off
 function ReaderStatistics:onResume()
     self.start_current_period = os.time()
-    self:resetVolatileStats(self.start_current_period)
+    self:onReadingResumed()
+end
+
+function ReaderStatistics:onReadingPaused()
+    if self:isDocless() or not self.settings.is_enabled then
+        return
+    end
+    if not self._reading_paused_ts then
+        self._reading_paused_ts = os.time()
+    end
+end
+
+function ReaderStatistics:onReadingResumed()
+    if self:isDocless() or not self.settings.is_enabled then
+        self._reading_paused_ts = nil
+        return
+    end
+    if self._reading_paused_ts then
+        -- Just add the pause duration to the current page start_time
+        local pause_duration = os.time() - self._reading_paused_ts
```

Comment:
I'd be nitpicking if I said that hilarity will ensue if time ever goes backward, because the whole plugin actually subtly breaks in those cases anyway, so it's in no way worse than before ;).